source/server/server.cc

```diff
@@ -276,6 +276,10 @@ void InstanceBase::updateServerStats() {
       enumToInt(Utility::serverState(initManager().state(), healthCheckFailed())));
   server_stats_->stats_recent_lookups_.set(
       stats_store_.symbolTable().getRecentLookups([](absl::string_view, uint64_t) {}));
+
+  // Update certificate stats for each context.
+  ssl_context_manager_->iterateContexts(static_cast<std::function<void(Envoy::Ssl::Context&)>>(
+      [&](Envoy::Ssl::Context& context) -> void { context.updateCertStats(); }));
```

Comment:
Is it necessary to poll for updates here? I think you can set the correct stats whenever the secret-update callback is run in the tls_context.